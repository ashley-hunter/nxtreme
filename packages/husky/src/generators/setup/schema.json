{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Setup",
  "title": "",
  "type": "object",
  "properties": {
    "precommit": {
      "type": "boolean",
      "description": "Setup a pre-commit hook",
      "default": true
    },
    "prepush": {
      "type": "boolean",
      "description": "Setup a pre-push hook",
      "default": true
    }
  },
  "required": []
}
