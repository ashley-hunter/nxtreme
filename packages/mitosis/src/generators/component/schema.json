{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Component",
  "title": "",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested).",
      "alias": "dir"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    }
  },
  "required": ["name", "project"]
}
