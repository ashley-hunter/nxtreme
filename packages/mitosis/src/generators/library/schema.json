{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Library",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?",
      "pattern": "^[a-zA-Z].*$"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "frameworks": {
      "type": "string",
      "description": "Comma separated list of frameworks to support.",
      "default": "angular, react, vue3"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "jest"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "publishable": {
      "type": "boolean",
      "description": "Create a publishable library."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library."
    },
    "importPath": {
      "type": "string",
      "description": "The library name used to import it, like `@myorg/my-awesome-lib`."
    },
    "component": {
      "type": "boolean",
      "description": "Generate a default component.",
      "default": true
    },
    "strict": {
      "type": "boolean",
      "description": "Whether to enable tsconfig strict mode or not.",
      "default": true
    },
    "setParserOptionsProject": {
      "type": "boolean",
      "description": "Whether or not to configure the ESLint `parserOptions.project` option. We do not do this by default for lint performance reasons.",
      "default": false
    },
    "standaloneConfig": {
      "description": "Split the project configuration into `<projectRoot>/project.json` rather than including it inside `workspace.json`.",
      "type": "boolean"
    },
    "skipTsConfig": {
      "type": "boolean",
      "default": false,
      "description": "Do not update `tsconfig.json` for development experience."
    }
  },
  "required": ["name"]
}
