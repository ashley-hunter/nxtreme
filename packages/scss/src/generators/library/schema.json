{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Library",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "importPath": {
      "type": "string",
      "description": "The library name used to import it, like `@myorg/my-awesome-lib`."
    },
    "publishable": {
      "type": "boolean",
      "description": "Create a publishable library."
    }
  },
  "required": ["name"]
}
