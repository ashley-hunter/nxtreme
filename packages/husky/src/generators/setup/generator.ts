import {
  addDependenciesToPackageJson,
  convertNxGenerator,
  formatFiles,
  GeneratorCallback,
  Tree,
  updateJson,
} from '@nrwl/devkit';
import { PackageJson } from 'nx/src/utils/package-json';
import { SetupGeneratorSchema } from './schema';

export async function setupGenerator(
  tree: Tree,
  options: SetupGeneratorSchema
): Promise<GeneratorCallback> {
  const installTask = addDependenciesToPackageJson(
    tree,
    {},
    {
      husky: '^8.0.1',
    }
  );

  updateJson<PackageJson>(tree, 'package.json', json => {
    json.scripts = {
      ...json.scripts,
      prepare: json.scripts?.prepare ? `${json.scripts.prepare} && husky install` : 'husky install',
    };

    return json;
  });

  if (options.precommit) {
    tree.write(
      '.husky/pre-commit',
      `#!/bin/sh
    . "$(dirname "$0")/_/husky.sh"

    nx=./node_modules/@nrwl/cli/bin/nx.js

    if [ -f "$nx" ]; then
      node $nx format:write
    fi`
    );
  }

  if (options.prepush) {
    tree.write(
      '.husky/pre-push',
      `#!/bin/sh
    . "$(dirname "$0")/_/husky.sh"

    nx=./node_modules/@nrwl/cli/bin/nx.js

    if [ -f "$nx" ]; then
      node "$nx" format:check
    fi`
    );
  }

  await formatFiles(tree);

  return installTask;
}

export default setupGenerator;

export const setupSchematic = convertNxGenerator(setupGenerator);
