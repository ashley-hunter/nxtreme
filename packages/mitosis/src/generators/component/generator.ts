import {
  convertNxGenerator,
  formatFiles,
  generateFiles,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import * as path from 'path';
import { ComponentGeneratorSchema } from './schema';

function addFiles(tree: Tree, options: ComponentGeneratorSchema) {
  const { sourceRoot } = readProjectConfiguration(tree, options.project);

  const componentDir = joinPathFragments(sourceRoot, getDirectory(options) ?? '');

  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(tree, path.join(__dirname, 'files'), componentDir, templateOptions);
}

export async function componentGenerator(tree: Tree, options: ComponentGeneratorSchema) {
  addFiles(tree, options);
  await formatFiles(tree);
}

function getDirectory(options: ComponentGeneratorSchema): string {
  const { fileName } = names(options.name);

  let baseDir: string;

  if (options.directory) {
    baseDir = options.directory;
  }

  return options.flat ? baseDir : joinPathFragments(baseDir, fileName);
}

export default componentGenerator;
export const componentSchematic = convertNxGenerator(componentGenerator);
