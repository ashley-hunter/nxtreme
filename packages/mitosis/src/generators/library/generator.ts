import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  TargetConfiguration,
  Tree,
  updateJson,
} from '@nrwl/devkit';
import {
  getRelativePathToRootTsConfig,
  getRootTsConfigPathInTree,
} from '@nrwl/workspace/src/utilities/typescript';
import * as path from 'path';
import { LibraryGeneratorSchema } from './schema';

interface NormalizedSchema extends LibraryGeneratorSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
  rootTsConfigPath: string;
}

function normalizeOptions(tree: Tree, options: LibraryGeneratorSchema): NormalizedSchema {
  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const projectRoot = `${getWorkspaceLayout(tree).libsDir}/${projectDirectory}`;
  const parsedTags = options.tags ? options.tags.split(',').map(s => s.trim()) : [];

  return {
    ...options,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
    rootTsConfigPath: getRelativePathToRootTsConfig(tree, projectRoot),
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    offsetFromRoot: offsetFromRoot(options.projectRoot),
    template: '',
  };
  generateFiles(tree, path.join(__dirname, 'files'), options.projectRoot, templateOptions);

  if (!options.publishable && !options.buildable) {
    tree.delete(`${options.projectRoot}/package.json`);
  }
}

export default async function (tree: Tree, options: LibraryGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);

  if (options.publishable === true && !options.importPath) {
    throw new Error(
      `For publishable libs you have to provide a proper "--importPath" which needs to be a valid npm package name (e.g. my-awesome-lib or @myorg/my-lib)`
    );
  }

  addProject(tree, normalizedOptions);
  updateBaseTsConfig(tree, normalizedOptions);
  addFiles(tree, normalizedOptions);

  await formatFiles(tree);
}

function addProject(tree: Tree, options: NormalizedSchema) {
  const targets: Record<string, TargetConfiguration> = {};

  if (options.publishable || options.buildable) {
    targets.build = {
      executor: '@nxtreme/mitosis:build',
      outputs: ['{options.outputPath}'],
      options: {
        outputPath: joinPathFragments('dist', options.projectRoot),
        tsConfig: joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
        entryFile: joinPathFragments(options.projectRoot, 'src', 'index.ts'),
      },
    };
  }

  addProjectConfiguration(
    tree,
    options.name,
    {
      root: options.projectRoot,
      projectType: 'library',
      sourceRoot: `${options.projectRoot}/src`,
      targets,
      tags: options.parsedTags,
    },
    options.standaloneConfig
  );
}

function updateBaseTsConfig(tree: Tree, options: NormalizedSchema) {
  updateJson(tree, getRootTsConfigPathInTree(tree), json => {
    const c = json.compilerOptions;
    c.paths = c.paths || {};
    delete c.paths[options.name];

    if (c.paths[options.importPath]) {
      throw new Error(
        `You already have a library using the import path "${options.importPath}". Make sure to specify a unique one.`
      );
    }

    const { libsDir } = getWorkspaceLayout(tree);

    c.paths[options.importPath] = [
      joinPathFragments(libsDir, `${options.projectDirectory}/src/index.ts`),
    ];

    return json;
  });
}
