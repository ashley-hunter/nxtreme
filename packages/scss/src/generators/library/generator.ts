import {
  addDependenciesToPackageJson,
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  readJsonFile,
  Tree,
  writeJsonFile,
} from '@nrwl/devkit';
import { LibraryGeneratorSchema } from './schema';

interface NormalizedSchema extends LibraryGeneratorSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

function normalizeOptions(
  tree: Tree,
  options: LibraryGeneratorSchema
): NormalizedSchema {
  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const projectRoot = `${getWorkspaceLayout(tree).libsDir}/${projectDirectory}`;
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  return {
    ...options,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    offsetFromRoot: offsetFromRoot(options.projectRoot),
    importPath: options.importPath ?? '',
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.projectRoot,
    templateOptions
  );
}

export default async function (
  tree: Tree,
  options: LibraryGeneratorSchema
): Promise<GeneratorCallback> {
  const normalizedOptions = normalizeOptions(tree, options);
  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.projectRoot,
    projectType: 'library',
    sourceRoot: `${normalizedOptions.projectRoot}/src`,
    targets: {
      build: {
        executor: '@nxtreme/scss:build',
        outputs: ['{options.outputPath}'],
        options: {
          entryFile: joinPathFragments(
            normalizedOptions.projectRoot,
            'src',
            'index.scss'
          ),
          outputPath: joinPathFragments('dist', normalizedOptions.projectRoot),
          sourceMap: true,
        },
      },
    },
    tags: normalizedOptions.parsedTags,
  });
  addFiles(tree, normalizedOptions);

  if (normalizedOptions.publishable && !normalizedOptions.importPath) {
    throw new Error(
      `For publishable libs you have to provide a proper "--importPath" which needs to be a valid npm package name (e.g. my-awesome-lib or @myorg/my-lib)`
    );
  }

  if (!normalizedOptions.publishable) {
    tree.delete(
      joinPathFragments(normalizedOptions.projectRoot, 'package.json')
    );
  }

  const installTask = addDependenciesToPackageJson(
    tree,
    {},
    {
      sass: '^1.54.4',
    }
  );

  addScssMapping(tree, normalizedOptions);

  await formatFiles(tree);

  return installTask;
}

function addScssMapping(tree: Tree, options: NormalizedSchema): void {
  let config: ScssConfig = {};

  if (tree.exists('/.scssrc')) {
    config = readJsonFile<ScssConfig>('/.scssrc');
  }

  config.paths = {
    ...config.paths,
    [options.importPath]: joinPathFragments(options.projectRoot, 'src'),
  };

  writeJsonFile('/.scssrc', config);
}

interface ScssConfig {
  paths?: Record<string, string>;
}
