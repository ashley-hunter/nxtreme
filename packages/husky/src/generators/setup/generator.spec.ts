import { readJson, Tree, updateJson } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { PackageJson } from 'nx/src/utils/package-json';
import { setupGenerator } from './generator';
import { SetupGeneratorSchema } from './schema';

describe('setup generator', () => {
  let appTree: Tree;
  const options: SetupGeneratorSchema = { precommit: true, prepush: true };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace();
  });

  it('should add the package', async () => {
    await setupGenerator(appTree, options);
    const json = readJson<PackageJson>(appTree, 'package.json');

    expect(json.devDependencies?.husky).toBe('^8.0.1');
  });

  it('should add the prepare script', async () => {
    await setupGenerator(appTree, options);
    const json = readJson<PackageJson>(appTree, 'package.json');

    expect(json.scripts?.prepare).toBe('husky install');
  });

  it('should add the prepare script to and existing prepare script', async () => {
    updateJson(appTree, 'package.json', json => {
      json.scripts = {
        prepare: 'nx dep-graph',
      };
      return json;
    });
    await setupGenerator(appTree, options);
    const json = readJson<PackageJson>(appTree, 'package.json');

    expect(json.scripts?.prepare).toBe('nx dep-graph && husky install');
  });

  it('should add the pre-commit script', async () => {
    await setupGenerator(appTree, options);
    expect(appTree.read('.husky/pre-commit', 'utf-8')).toMatchSnapshot();
  });

  it('should add the pre-push script', async () => {
    await setupGenerator(appTree, options);
    expect(appTree.read('.husky/pre-push', 'utf-8')).toMatchSnapshot();
  });
});
