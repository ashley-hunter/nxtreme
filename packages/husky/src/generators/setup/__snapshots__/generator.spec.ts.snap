// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup generator should add the pre-commit script 1`] = `
"#!/bin/sh
    . \\"$(dirname \\"$0\\")/_/husky.sh\\"

    nx=./node_modules/@nrwl/cli/bin/nx.js

    if [ -f \\"$nx\\" ]; then
      node $nx format:write
    fi"
`;

exports[`setup generator should add the pre-push script 1`] = `
"#!/bin/sh
    . \\"$(dirname \\"$0\\")/_/husky.sh\\"

    nx=./node_modules/@nrwl/cli/bin/nx.js

    if [ -f \\"$nx\\" ]; then
      node \\"$nx\\" format:check
    fi"
`;
